library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java

```diff
@@ -165,8 +120,31 @@ public boolean isProviderEnabled() {
 
     @Override
     public void close() {
-        disconnectGoogleApiClient();
-
+        stopLocationUpdates();
         super.close();
     }
+
+    class AndroidLocationCallback extends LocationCallback {
+
+        @Override
+        public void onLocationResult(LocationResult locationResult) {
+            if (locationResult == null) {
+                return;
+            }
+
+            Location latestLocation = null;
+            for (Location location : locationResult.getLocations()) {
+                if (latestLocation == null || location.getTime() > latestLocation.getTime()) {
+                    latestLocation = location;
+                }
+            }
+
+            if (latestLocation != null) {
+                lastLocation = latestLocation;
+                if (getLocationListener() != null) {
```

Comment:
Why not combine this if statement with the one enclosing it and perform this `lastLocation = latestLocation` after calling the listener?

Reply:
So if `getLocationListener()` returns null for some reason, we will not update `lastLocation`?

Reply:
does it need to be updated in this case?

Reply:
Yea

Reply:
I think a better comment to explain why would be helpful.

Reply:
I don't think that's necessary.